src/execution/operator/persistent/physical_delete.cpp

```diff
@@ -64,22 +64,66 @@ SinkResultType PhysicalDelete::Sink(ExecutionContext &context, DataChunk &chunk,
 	auto &transaction = DuckTransaction::Get(context.client, table.db);
 	auto &row_ids = chunk.data[row_id_index];
 
-	vector<StorageIndex> column_ids;
-	for (idx_t i = 0; i < table.ColumnCount(); i++) {
-		column_ids.emplace_back(i);
-	};
-	auto fetch_state = ColumnFetchState();
-
 	lock_guard<mutex> delete_guard(g_state.delete_lock);
 	if (!return_chunk && !g_state.has_unique_indexes) {
 		g_state.deleted_count += table.Delete(*l_state.delete_state, context.client, row_ids, chunk.size());
 		return SinkResultType::NEED_MORE_INPUT;
 	}
 
+	auto types = table.GetTypes();
+	auto to_be_fetched = vector<bool>(types.size(), false);
```

Comment:
Nit: `auto to_be_fetched = vector<bool>(types.size(), return_chunk);`, and then remove `to_be_fetched[i] = true;`.